src/actions/importing/abstract-aioseo-settings-importing-action.php

```diff
@@ -357,4 +356,48 @@ public function simple_import( $meta_data ) {
 	public function simple_boolean_import( $meta_data ) {
 		return $meta_data;
 	}
+
+	/**
+	 * Retrieves the noindex setting set globally in AIOSEO.
+	 *
+	 * @return bool Whether global robot settings give a noindex or not.
+	 */
+	public function get_global_noindex() {
+		$aioseo_settings = \json_decode( \get_option( 'aioseo_options', [] ), true );
+		if ( empty( $aioseo_settings ) || ! isset( $aioseo_settings['searchAppearance']['advanced']['globalRobotsMeta'] ) ) {
+			return false;
+		}
+
+		$global_robot_settings = $aioseo_settings['searchAppearance']['advanced']['globalRobotsMeta'];
+		if ( ! isset( $global_robot_settings['default'] ) || $global_robot_settings['default'] === true ) {
+			return false;
+		}
+
+		return isset( $global_robot_settings['noindex'] ) ? $global_robot_settings['noindex'] : false;
+	}
+
+	/**
+	 * Imports the noindex setting, taking into consideration whether they defer to global defaults.
+	 *
+	 * @param bool  $noindex The noindex of the type, without taking into consideration whether the type defers to global defaults.
+	 * @param array $mapping The mapping of the setting we're working with.
+	 *
+	 * @return bool The noindex setting.
+	 */
+	public function import_noindex( $noindex, $mapping ) {
+		$aioseo_settings = \json_decode( \get_option( $mapping['option_name'], [] ), true );
+
+		// Let's check first if it defers to global robot settings.
+		if ( empty( $aioseo_settings ) || ! isset( $aioseo_settings['searchAppearance'][ $mapping['type'] ][ $mapping['subtype'] ]['advanced']['robotsMeta']['default'] ) ) {
+			return false;
```

Comment:
Wouldn't it make more sense to also return `$noindex` here?

Reply:
indeed.
```suggestion
			return $noindex;
```